core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2909,6 +2911,37 @@ private File getRootDirFor(String name) {
         return new File(new File(getRootDir(),"jobs"), name);
     }
 
+    public static String urlEncodeMatrixConfigString(String fullName) {
+        String html_escaped_string = "";
+        StringTokenizer parameter_tokens = new StringTokenizer(fullName, "=");
+        while (parameter_tokens.hasMoreTokens()) {
+            String before_equal = parameter_tokens.nextToken();
+            String after_equal = "";
+            if (parameter_tokens.hasMoreTokens()) {
+                after_equal = parameter_tokens.nextToken();
+            } else {
+                after_equal = before_equal;
+                before_equal = "";
+            }
+            try {
+                //encode only after =. If you escape before_equal then it will break logic
+                html_escaped_string = html_escaped_string + before_equal + "=" + URLEncoder.encode(after_equal, "UTF-8");
```

Comment:
Is this working for the case where there are no more token and you set before_equal to an empty string? It seems it would be adding ='something' without anything before the equal.